### PR TITLE
Renamed docs/developer-guide.md to docs/CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by eranco74**
Added Pull Request, Code review and Best Practices sections to CONTRIBUTING.md

This should help new contributors to send a PR
also it will fix the broken link in the github checks under the DCO details.

```release-note
NONE
```

### CODE REVIEW

File: docs/deployment-local-cluster.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ the [deployment on arbitrary cluster guide](deployment-arbitrary-cluster.md).
 
 Start your local cluster. If you want to use OpenShift instead of Kubernetes or
 a different amount of nodes, check the
-[development guide](developer-guide.md#local-cluster).
+[development guide](CONTRIBUTING.md#local-cluster).
```
Comment (phoracek) [inline docs/deployment-local-cluster.md]: Would not work, should be `../CONTRIBUTING.md`

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+kubernetes-nmstate specific github workflow:\
+The first difference you'll see is that a bot will begin applying structured labels to your PR.
+
+The bot may also make some helpful suggestions for commands to run in your PR to facilitate review.
+These /command options can be entered in comments to trigger auto-labeling and notifications. Refer to its command reference documentation.
+
+Common new contributor PR issues are:
+
+* Missing DCO sign-off:
+Developers Certificate of Origin (DCO) Sign-off is a requirement for getting patches into the project (see [Developers Certificate of Origin](https://developercertificate.org/)).\
```
Comment (phoracek) [inline docs/CONTRIBUTING.md]: You don't need to enforce new line using `\`. Could can use normal paragraph and indent the following lines (at least I believe so)

File: docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
+# Design
+
+The system is implemented as an k8s operator using the [operator-sdk](https://github.com/operator-framework/operator-sdk)
+but is deployed as a DaemonSet instead of Deployment with [filtering](https://github.com/operator-framework/operator-sdk/blob/master/doc/user/event-filtering.md) only events for the DaemonSet pod node.
+
+There are two controllers one for [Node](https://godoc.org/k8s.io/api/core/v1#Node)
```
Comment (phoracek) [inline docs/CONTRIBUTING.md]: This is painfully outdated :/ Since you just moved this part, don't worry fixing it.

File: docs/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+# Development
+
+[Development guide](developer-guide-commands.md) is a go-to reference point for
+development helper commands, building, testing, container images and local
+cluster.
+
+# Open a Pull Request
+Kubernetes-nmstate generally follows the [standard github pull request process](https://gist.github.com/Chaser324/ce0505fbed06b947d962), but there is a layer of additional specific differences:
+
+kubernetes-nmstate specific github workflow:\
```
Comment (phoracek) [inline docs/CONTRIBUTING.md]: Please don't enforce newlines.  It is fine to keep it on a single line.

File: docs/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+Developers Certificate of Origin (DCO) Sign-off is a requirement for getting patches into the project (see [Developers Certificate of Origin](https://developercertificate.org/)).\
+You can "sign" this certificate by including a line in the git commit of "Signed-off-by: Legal Name <email-address>".\
+If you forgot to add the sign-off, you can also amend your commit with the sign-off:\
+`git commit --amend -s`.
+
+## Code Review
+
+To make it easier for your PR to receive reviews, consider the reviewers will need you to:
+
+* Follow the project [coding conventions](https://github.com/golang/go/wiki/CodeReviewComments).
```
Comment (phoracek) [inline docs/CONTRIBUTING.md]: Please be consistent with the list syntax, either `-` or `*`.

File: docs/CONTRIBUTING.md (hunk 3 above)
Comment (eranco74) [reply docs/CONTRIBUTING.md]: I guess this is a part of the original file

**Comment by phoracek** [comment]
/lgtm
/approve
/override pull-kubernetes-nmstate-e2e-k8s
/override

Thanks!

**Pull request merged**